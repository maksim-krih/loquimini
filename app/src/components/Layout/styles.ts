import {createUseStyles} from 'react-jss'

export const useStyles = createUseStyles({
  trigger: {
    padding: "0 24px",
    fontSize: 18,
    lineHeight: "64px",
    cursor: "pointer",
    transition: "color 0.3s",
    "&:hover": {
      color: "#1890ff"
    }
  },
  logo: {
    height: 32,
    margin: 16,
    background: "rgba(255, 255, 255, 0.3)"
  },
  layout: {
    background: "#fff"
  },
  header: {
    padding: 0
  },
  content: {
    margin: '24px 16px',
    padding: 24,
    minHeight: 280,
  }
})