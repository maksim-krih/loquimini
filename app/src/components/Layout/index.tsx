import React, { FC } from 'react';
import { useToggle } from 'ahooks';
import { Avatar, Dropdown, Layout as AntLayout, Menu } from 'antd';
import {
  MenuUnfoldOutlined,
  MenuFoldOutlined,
  UserOutlined,
  MoreOutlined,
  HomeOutlined,
  ApartmentOutlined,
  ToolOutlined,
  DollarOutlined
} from '@ant-design/icons';
import { IProps } from './types';
import { useStyles } from './styles';
import { Typography } from 'antd';
import { AuthService } from '../../services';
import { useHistory } from 'react-router';
import { RouterPaths } from '../../consts';

const { Title, Text } = Typography;
const { Header, Sider, Content } = AntLayout;

const Layout: FC<IProps> = (props: IProps) => {
  const classes = useStyles();
  const history = useHistory();
  const user = AuthService.User;

  const [collapsed, { toggle }] = useToggle(false);

  const onSignOut = () => {
    AuthService.SignOut();
    history.push(RouterPaths.Login);
  }

  const profileMenu = (
    <Menu>
      <Menu.Item>
        Profile
      </Menu.Item>
      <Menu.Item onClick={onSignOut}>
        Sign Out
      </Menu.Item>
    </Menu>
  );
  
    return (
      <AntLayout className={classes.layoutContainer}>
        <Sider trigger={null} collapsible collapsed={collapsed}>
          <div className={classes.logo}>
          <Title level={3}>{!collapsed ? "Loquimini" : "L"}</Title>
          </div>
          <Menu theme="dark" mode="inline">
            {AuthService.IsAdmin ? (
              <>
                <Menu.Item key="1" icon={<ApartmentOutlined />} onClick={() => history.push(RouterPaths.HouseList)}>
                  Houses
                </Menu.Item>
                <Menu.Item key="2" icon={<UserOutlined />} onClick={() => history.push(RouterPaths.UserList)}>
                  Users
                </Menu.Item>
                <Menu.Item key="3" icon={<ToolOutlined />} onClick={() => history.push(RouterPaths.Triggers)}>
                  Triggers
                </Menu.Item>
              </>
            ) : (
              <>
                <Menu.Item key="1" icon={<HomeOutlined />} onClick={() => history.push(RouterPaths.Dashboard)}>
                  Dashboard
                </Menu.Item>
                <Menu.Item key="2" icon={<DollarOutlined />} onClick={() => history.push(RouterPaths.Receipts)}>
                  Receipts
                </Menu.Item>
              </>
            )}
          </Menu>
        </Sider>
        <AntLayout className={classes.layout}>
          <Header className={classes.header}>
            {collapsed ? 
              <MenuUnfoldOutlined className={classes.trigger} onClick={() => toggle()}/> 
              : 
              <MenuFoldOutlined  className={classes.trigger} onClick={() => toggle()}/>
            }
            <div className={classes.profile}>
              <Avatar className={classes.user} icon={<UserOutlined />}/>
              <Text >{`${user.firstName} ${user.lastName}`.trim()}</Text>
              <Dropdown overlay={profileMenu} trigger={['click']}>
                <MoreOutlined />
              </Dropdown>
            </div>
          </Header>
          <Content
            className={classes.contentContainer}
          >
            <div className={classes.content}>
              {props.children}
            </div>
          </Content>
        </AntLayout>
      </AntLayout>
    );
};

export default Layout;