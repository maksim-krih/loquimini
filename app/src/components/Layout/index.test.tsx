import Layout from ".";
import { shallow } from "enzyme";

const setUp = () => shallow(<Layout />);

describe("Layout", () => {
  it("should render Layout Component", () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
});