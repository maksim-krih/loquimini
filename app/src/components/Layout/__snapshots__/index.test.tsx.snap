// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Layout Component 1`] = `
<Layout
  className="layoutContainer-0-2-7"
>
  <Sider
    collapsed={false}
    collapsible={true}
    trigger={null}
  >
    <div
      className="logo-0-2-2"
    >
      <img
        alt="logo"
        src="logo.png"
        style={
          Object {
            "filter": "invert(1)",
            "marginRight": 8,
          }
        }
      />
      <Title
        level={3}
      >
        Loquimini
      </Title>
    </div>
    <Menu
      mode="inline"
      theme="dark"
    >
      <MenuItem
        icon={<ForwardRef(HomeOutlined) />}
        key="1"
        onClick={[Function]}
      >
        Dashboard
      </MenuItem>
      <MenuItem
        icon={<ForwardRef(DollarOutlined) />}
        key="2"
        onClick={[Function]}
      >
        Receipts
      </MenuItem>
    </Menu>
  </Sider>
  <Layout
    className="layout-0-2-3"
  >
    <Header
      className="header-0-2-4"
    >
      <ForwardRef(MenuFoldOutlined)
        className="trigger-0-2-1"
        onClick={[Function]}
      />
      <div
        className="profile-0-2-8"
      >
        <Avatar
          className="user-0-2-9"
          icon={<ForwardRef(UserOutlined) />}
          shape="circle"
          size="default"
        />
        <Text />
        <Dropdown
          mouseEnterDelay={0.15}
          mouseLeaveDelay={0.1}
          overlay={
            <Menu>
              <MenuItem>
                Profile
              </MenuItem>
              <MenuItem
                onClick={[Function]}
              >
                Sign Out
              </MenuItem>
            </Menu>
          }
          trigger={
            Array [
              "click",
            ]
          }
        >
          <ForwardRef(MoreOutlined) />
        </Dropdown>
      </div>
    </Header>
    <Content
      className="contentContainer-0-2-5"
    >
      <div
        className="content-0-2-6"
      />
    </Content>
  </Layout>
</Layout>
`;
